### ISSUE TITLE
Passing false to certain validators should mean they are not enforced (e.g. `required: false`)

### ISSUE THREAD

**Issue by cspotcode** [balderdashy/anchor#70]
Related to https://github.com/balderdashy/waterline/issues/867

Many validators are implemented such that, if the corresponding property is present, the validator will be enforced regardless of value.  However, this behavior is non-intuitive for certain validations.

For example:
```
anchor('').to({
    required: false,
    uuid: true
});
```
Intuitively, this means the value is not required and the "required" validator should always pass.
However, actual behavior is identical to `required: true`: empty values aren't allowed.

For other validators it might be ambiguous.  For example, `empty: true` makes sense, but what is the meaning of `empty: false`?
A. value can be empty or not; doesn't matter
B. value is prohibited from being empty; must be non-empty

I propose that, for validators that don't accept a value, passing them a falsey value means they don't execute.  This would be a breaking change, but I think that's fine.

**Comment by sailsbot** [comment]
Thanks for posting, @cspotcode.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
